lua/delaytrain/init.lua

```diff
@@ -83,10 +83,18 @@ function M.enable()
             table.insert(mode_array, mode)
         end
         for _, key in ipairs(keys) do
+            -- Check that keys haven't been remapped (e.g. hjkl to dtrn)
+            local keypress = ""
+            local remapped = vim.fn.maparg(key, mode_array[1])
```

Comment:
If a mapping is set to a lua function, `vim.fn.maparg` will return a string representation of the function (ex: "<Lua function 240>"). This string will be interpreted as literal keypresses. So custom mappings to lua functions won't work, which I think is fine for now. But if the user accidentally runs `:DelayTrainEnable` while DelayTrain is already enabled, the mappings set by DelayTrain are returned as these string representations, which causes issues. 

There's a good trick to ignore mappings to lua functions here and still get the remapped keypress. `vim.fn.maparg` can optionally return a table with a few different values. One of these is `rhs`, which will only return the remapped value if it's an actual keypress. So you should be able to change the above line to the following:

```lua
local remapped = vim.fn.maparg(key, mode_array[1], false, true).rhs
```

Reply:
Thanks for all the feedback, I appreciate the opportunity. Working on all this now. It seemed like I needed to make this change as well after adopting your change:

Reply:
Just realized that the `if remapped == ""` check isn't doing anything anymore. In my original line, the function returned `""` if there was no remap (as described in the second paragraph of the `:h` doc for `maparg`, "When there is no mapping for `{name}`, an empty String is returned if {dict} is FALSE..." ), but now I understand why using `.rhs` makes more sense even if we weren`t trying to ignore mappings to lua functions. So the updated lines, I think, should be

```lua
local remapped = vim.fn.maparg(key, mode_array[1], false, true).rhs
if remapped == nil then
```